src/core/gui/dialog/LatexDialog.cpp

```diff
@@ -13,26 +13,140 @@
-    this->texBox = get("texView");
+#ifdef USE_GTK_SOURCEVIEW
+#include <gtksourceview/gtksource.h>
+#endif
+
+#include "control/settings/Settings.h"
+#include "util/StringUtils.h"
+
+// Default background color of the preview.
+constexpr Color DEFAULT_PREVIEW_BACKGROUND{0xFFFFFFFF};
+
+// Callbacks for gtk to render the dialog's preview.
+extern "C" {
+/**
+ * @brief Called on 'size-allocate' signal.
+ * @param widget is the event's target.
+ * @param allocation provides the region given to the widget.
+ */
+static void resizePreviewCallback(GtkWidget* widget, GdkRectangle* allocation, gpointer _unused);
+}
+
+LatexDialog::LatexDialog(GladeSearchpath* gladeSearchPath, const LatexSettings& settings):
+        GladeGui(gladeSearchPath, "texdialog.glade", "texDialog"), previewBackgroundColor{DEFAULT_PREVIEW_BACKGROUND} {
+    GtkContainer* texBoxContainer = GTK_CONTAINER(get("texBoxContainer"));
+
+    this->cssProvider = gtk_css_provider_new();
+#ifdef USE_GTK_SOURCEVIEW
+    this->texBox = gtk_source_view_new();
+#else
+    this->texBox = gtk_text_view_new();
+#endif
+
+    gtk_widget_set_name(this->texBox, "texBox");
+    gtk_container_add(texBoxContainer, this->texBox);
+    gtk_text_view_set_editable(GTK_TEXT_VIEW(this->texBox), true);
+    gtk_widget_show_all(GTK_WIDGET(texBoxContainer));
+
     this->textBuffer = gtk_text_view_get_buffer(GTK_TEXT_VIEW(this->texBox));
     this->texTempRender = get("texImage");
+    gtk_widget_set_name(GTK_WIDGET(this->texTempRender), "texImage");
+
+#ifdef USE_GTK_SOURCEVIEW
+    // We own neither the languageManager, the styleSchemeManager, nor the sourceLanguage.
+    // Do not attempt to free them.
+    GtkSourceStyleSchemeManager* styleSchemeManager = gtk_source_style_scheme_manager_get_default();
+    GtkSourceLanguageManager* lm = gtk_source_language_manager_get_default();
+
+    // Select the TeX highlighting scheme
+    GtkSourceLanguage* lang = gtk_source_language_manager_guess_language(lm, "file.tex", NULL);
+    std::string styleSchemeId = settings.sourceViewThemeId;
+    GtkSourceStyleScheme* styleScheme =
+            gtk_source_style_scheme_manager_get_scheme(styleSchemeManager, styleSchemeId.c_str());
+
+    if (settings.sourceViewSyntaxHighlight) {
+        gtk_source_buffer_set_language(GTK_SOURCE_BUFFER(this->textBuffer), lang);
+    }
+
+    gtk_source_view_set_auto_indent(GTK_SOURCE_VIEW(this->texBox), settings.sourceViewAutoIndent);
+    gtk_source_view_set_indent_on_tab(GTK_SOURCE_VIEW(this->texBox), settings.sourceViewAutoIndent);
+
+    gtk_source_view_set_show_line_numbers(GTK_SOURCE_VIEW(this->texBox), settings.sourceViewShowLineNumbers);
+
+    if (styleScheme) {
+        gtk_source_buffer_set_style_scheme(GTK_SOURCE_BUFFER(this->textBuffer), styleScheme);
+    }
+#endif
+
+    // Enable/disable word-wrap.
+    GtkWrapMode texBoxWrapMode = GTK_WRAP_NONE;
+    if (settings.editorWordWrap) {
+        texBoxWrapMode = GTK_WRAP_WORD_CHAR;
+    }
```

Comment:
```suggestion
    GtkWrapMode texBoxWrapMode = settings.editorWordWrap ? GTK_WRAP_WORD_CHAR : GTK_WRAP_NONE;
```
looks simpler to me.